File: sdk-java/src/main/java/com/primust/sdk/CheckResult.java
package com.primust.sdk;

/**
 * Result of a governance check execution.
 */
public enum CheckResult {
    PASS("pass"),
    FAIL("fail"),
    ERROR("error"),
    SKIPPED("skipped"),
    DEGRADED("degraded"),
    OVERRIDE("override"),
    NOT_APPLICABLE("not_applicable");

    private final String value;

    CheckResult(String value) {
        this.value = value;
    }

    public String value() {
        return value;
    }

    @Override
    public String toString() {
        return value;
    }
}

File: sdk-csharp/src/Primust/CheckResult.cs
namespace Primust;

/// <summary>
/// Result of a governance check execution.
/// </summary>
public enum CheckResult
{
    Pass,
    Fail,
    Error,
    Skipped,
    Degraded,
    Override,
    NotApplicable
}

internal static class CheckResultExtensions
{
    public static string ToWireValue(this CheckResult result) => result switch
    {
        CheckResult.Pass => "pass",
        CheckResult.Fail => "fail",
        CheckResult.Error => "error",
        CheckResult.Skipped => "skipped",
        CheckResult.Degraded => "degraded",
        CheckResult.Override => "override",
        CheckResult.NotApplicable => "not_applicable",
        _ => "unknown"
    };
}
